File: linkedlist/circular_linked_list/circlar_linked_list.py
##Implmentation of Circular linked list

class node:
    def __init__(self ,data):
        self.data = data
        self.next = None

class linked_list:
    def __init__(self):
        self.HEAD = None

def print_linked_list(Head):
    while Head != None:
        print(Head.data ,"\t")

        Head = Head.next


if __name__ == '__main__':
    link_list = linked_list()

    one = node(0)
    two = node(1)
    three = node(2)
    four = node(3)

    one.next = two
    two.next = three
    three.next = four

    link_list.HEAD = one

    four.next = link_list.HEAD

    ##This will result in infinite loop which clearly states that one we reach at end of the linked list we will start traversing it from head again
    print_linked_list(link_list.HEAD)

File: linkedlist/circular_linked_list/circlar_linked_list.c
#include <stdio.h>
#include <stdlib.h>

struct node{
    int data;
    struct node *next;
};

void insert_at_begining(struct node **head_ref ,int data){
    struct node *tmp_node = (struct node *)malloc(sizeof(struct node));

    tmp_node->data  = data;
    tmp_node->next = (*head_ref);

    (*head_ref) = tmp_node;
}

void print_linked_list(struct node *head){
    while(head != NULL){
        printf("%d\t" ,head->data);

        head = head->next;
    }
}

int main(){

    struct node *HEAD;

    insert_at_begining(&HEAD ,0);
    insert_at_begining(&HEAD ,1);
    insert_at_begining(&HEAD ,2);
    insert_at_begining(&HEAD ,3);

    print_linked_list(HEAD);

    return 0;
}